ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ImageSaver.kt

```diff
@@ -0,0 +1,198 @@
+    ): Result<Unit> = withContext(dispatcher) {
+        contentValues.put(Media.IS_PENDING, true)
+
+        val uri = context.contentResolver.insert(MediaStore.Downloads.EXTERNAL_CONTENT_URI, contentValues)
+            ?: return@withContext Result.failure(CheckoutException("Error when saving Bitmap as an image: URI is null"))
+
+        return@withContext try {
+            val outputStream = context.contentResolver.openOutputStream(uri)
+                ?: return@withContext Result.failure(CheckoutException("Output stream is null"))
+
+            contentValues.put(Media.IS_PENDING, false)
+            context.contentResolver.update(uri, contentValues, null, null)
+
+            bitmap.compress(CompressFormat.PNG, PNG_QUALITY, outputStream)
+            outputStream.close()
+
+            Logger.d(TAG, "Bitmap successfully saved as an image")
+            Result.success(Unit)
+        } catch (e: FileNotFoundException) {
+            Result.failure(CheckoutException("File not found: ", e))
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    private suspend fun saveImageApi28AndBelow(
+        context: Context,
+        permissionHandler: PermissionHandler,
+        bitmap: Bitmap,
+        contentValues: ContentValues,
+    ): Result<Unit> = withContext(dispatcher) {
+        if (permissionHandler.checkPermission(context, REQUIRED_PERMISSION) == true) {
+            saveImageApi28AndBelowWhenPermissionGranted(bitmap, contentValues)
+        } else {
+            Result.failure(PermissionRequestException("The $REQUIRED_PERMISSION permission is denied"))
+        }
+    }
+
+    @RequiresPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+    private suspend fun saveImageApi28AndBelowWhenPermissionGranted(
+        bitmap: Bitmap,
+        contentValues: ContentValues
+    ): Result<Unit> = withContext(dispatcher) {
+        val fileName = contentValues.getAsString(Media.DISPLAY_NAME)
+        val filePath = contentValues.getAsString(Media.RELATIVE_PATH)
+        val imageFileFolder =
+            Environment.getExternalStoragePublicDirectory(filePath)
+        if (!imageFileFolder.exists()) {
+            imageFileFolder.mkdirs()
+        }
+        val imageFile = File(imageFileFolder, fileName)
+
+        return@withContext try {
+            val outputStream = FileOutputStream(imageFile)
+
+            bitmap.compress(CompressFormat.PNG, PNG_QUALITY, outputStream)
+            outputStream.close()
+
+            Logger.d(TAG, "Bitmap successfully saved as an image")
+            Result.success(Unit)
+        } catch (e: FileNotFoundException) {
+            Result.failure(CheckoutException("File not found: ", e))
+        } catch (e: SecurityException) {
+            Result.failure(CheckoutException("Security violation: ", e))
+        }
+    }
+
+    private fun String.toURL(): URL? {
+        return try {
+            URL(this)
+        } catch (e: MalformedURLException) {
+            Logger.e(TAG, "toURL: $e")
```

Comment:
I know this code comes from the old implementation, but could you maybe improve this message?